### DISCUSSION TITLE
External Authentication - Connecting an External Idp using external authentication hook

### DISCUSSION THREAD

**Discussion by VihangiPatterson** [drakkan/sftpgo#1824] [Q&A]
Hi,
I’m trying to connect an external IDP to authenticate users using our IDP credentials in SFTPGo. I successfully used keyboard-interactive authentication, but when I added TOTP and OTP options, it failed.

Now, I’m trying external authentication. I followed the documentation but couldn’t find a clear explanation. I tested with a webhook URL and updated the sftpgo.json configuration, but it doesn’t seem to send any requests or work.

Could you please provide some guidance on how to set up external authentication with SFTPGo?

Thank you!

**Comment by RickeyWard** [comment]
If the 'external_auth_hook' setting inside of 'data_provider' in the config json file is a URL, it will post to that url, but only if there's no other authentication method in-front of it. so the ldap auth plugin for example would need to be disabled for it to work. It's not obvious but the environment variable would be 'SFTPGO_DATA_PROVIDER__EXTERNAL_AUTH_HOOK'
